models/governance/arbitrum/governance_arbitrum_snapshot_approval_proposals.sql

```diff
@@ -0,0 +1,42 @@
+{{ config(alias = 'snapshot_approval_proposals'
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,schema = 'governance_arbitrum'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['proposal_id']
+    ,post_hook='{{ expose_spells(\'["arbitrum"]\',
+                                      "sector",
+                                      "governance",
+                                    \'["ARDev097"]\') }}'
+    )
+}}
+
+SELECT
+  dp.id AS proposal_id,
+  dp.title AS proposal_title,
+  CONCAT(CAST(SUBSTRING(dp.body, 1, 40) AS VARCHAR), CAST('..' AS VARCHAR)) AS proposal_description,
+  '<a href="https://snapshot.org/#/arbitrumfoundation.eth/proposal/' || CAST(dp.id AS varchar) || '" target="_blank">To Read More</a>' AS proposal_link,
+  dp."type" AS proposal_type,
+  dp.choices AS proposal_choice,
+  dv.voter AS voter_address,
+  dv.vp AS voting_weightage,
+  dv.choice as voting_choice,
+  TRY_CAST(TRIM(TRIM(t.value, '[]'), ' ') AS INTEGER) AS voting_choice_separated,
+  ELEMENT_AT(
+    SPLIT(TRIM(TRIM(dp.choices, '[]'), ' '), ', '),
+    TRY_CAST(TRIM(TRIM(t.value, '[]'), ' ') AS INTEGER)
+  ) AS separated_voting_choice_name,
+  dv.reason AS voting_reason,
+  FROM_UNIXTIME(dv.created) AS voting_timestamp,
+  YEAR(FROM_UNIXTIME(dv.created)) AS voting_year,
+  MONTH(FROM_UNIXTIME(dv.created)) AS voting_month,
+  DAY_OF_MONTH(FROM_UNIXTIME(dv.created)) AS voting_date
+FROM {{ source('dune.shot','dataset_proposals_view') }} AS dp
+JOIN {{ source('dune.shot','dataset_votes_view') }} AS dv
```

Comment:
Hi @jeff-dude,
do you know how to use community uploaded data as source in the spellbook?
https://dune.com/queries/3490075?category=third_party_data&catalog=dune&schema=shot&id=01HR26V1S4TK4D0RNFZ6E4QCWD

Reply:
Hello @Hosuke ,
Thank you for the suggestion, but when using these schemas also I got the error.

![image](https://github.com/duneanalytics/spellbook/assets/147229733/c3ac7bb3-789e-4334-9c3d-409ad407c717)

Reply:
Sorry ser, my suggestion does not work.
We may need to find a new solution.

Reply:
this is now unblocked, you can pass a `database` arg to the source jinja and override the default with `dune` or whatever is needed
for an example, please see `tokens_erc20.sql`